java/core/src/java/org/apache/orc/OrcConf.java

```diff
@@ -118,6 +118,11 @@ public enum OrcConf {
       "If the number of distinct keys in a dictionary is greater than this\n" +
           "fraction of the total number of non-null rows, turn off \n" +
           "dictionary encoding.  Use 1 to always use dictionary encoding."),
+  DICTIONARY_MAX_SIZE_IN_BYTES("orc.dictionary.maxSizeInBytes",
+      "hive.exec.orc.dictionary.maxSizeInBytes",
+      16 * 1024 * 1024,
+      "If the total size of the dictionary is greater than this\n" +
+          ", turn off dictionary encoding.  Use 1 to always use dictionary encoding."),
```

Comment:
> Use 1 to always use dictionary encoding.

It seems that this logic is not implemented.